File: semestre-1/algoritmos/lista-1/c/MR-Alg-01-Ex-11.c
#define _USE_MATH_DEFINES
#include <math.h>
#include <stdio.h>

#define PI 3.14159265358979323846

void main() {
   double t1, g1, t2, g2;

   printf("Latitude 1: ");
   scanf("%lf", &t1);

   printf("Longitude 1: ");
   scanf("%lf", &g1);

   printf("Latitude 2: ");
   scanf("%lf", &t2);

   printf("Longitude 2: ");
   scanf("%lf", &g2);

   // Converte para radiano
   t1 = t1 * PI / 180;
   g1 = g1 * PI / 180;
   t2 = t2 * PI / 180;
   g2 = g2 * PI / 180;

   double distancia = 6371.01 * acos(sin(t1) * sin(t2) + cos(t1) * cos(t2) * cos(g1 - g2));
   printf("Distância dos dois pontos: %.3lf km\n", distancia);
}

File: semestre-1/algoritmos/MR-Alg-01/.py/MR-Alg-01-Ex-11.py
from math import *

t1 = radians(float(input("Latitude 1: ")))
g1 = radians(float(input("Longitude 1: ")))
t2 = radians(float(input("Latitude 2: ")))
g2 = radians(float(input("Longitude 2: ")))

distancia = 6371.01 * acos(sin(t1) * sin(t2) + cos(t1) * cos(t2) * cos(g1 - g2))

print(f"Distância dos dois pontos: {distancia} km")